common/persistence/sql/execution_util.go

```diff
@@ -1152,3 +1161,28 @@ func updateExecution(
 
 	return nil
 }
+
+func workflowExecutionStateFromCurrentExecutionsRow(
```

Comment:
shall we change other places that uses current execution row to use the blob instead? e.g. GetCurrentExecution in sql/execution.go.

we still do double write so doesn't really matter right now.

Reply:
I thought about the same, but I'll check and see if it's gonna be an improvement or not in another PR.